extensions/wikia/Search/classes/ResultSet/Factory.php

```diff
@@ -3,35 +3,48 @@
  * Class definition for \Wikia\Search\ResultSet\Factory
  */
 namespace Wikia\Search\ResultSet;
-use \Solarium_Result_Select, \Solarium_Result_Select_Empty, \WikiaSearchConfig;
+
+use \Solarium_Result_Select, \Solarium_Result_Select_Empty, \Wikia\Search\Config;
+
 /**
  * A factory for instantiating search result sets.
  * This lets us make something else responsible for determining which instance to select.
+ *
  * @author relwell
  * @package Search
  * @subpackage ResultSet
  */
-class Factory
-{
+class Factory {
 	/**
 	 * Inspects a dependency container and selects the appropriate result set based on what is set.
+	 *
 	 * @param DependencyContainer $container
-	 * @todo return type hinting for DependencyContainer to constructor when we have a better solution than WikiaMockProxy for testing. (or a dependencycontainerfactory, yuck)
+	 *
+	 * @todo return type hinting for DependencyContainer to constructor when we have a better solution than
+	 *     WikiaMockProxy for testing. (or a dependencycontainerfactory, yuck)
 	 * @throws \Exception
 	 * @return \Wikia\Search\ResultSet\AbstractResultSet
 	 */
 	public function get( $container ) {
 		$searchConfig = $container->getConfig();
 		if ( $searchConfig === null ) {
-			throw new \Exception( 'An instance of Wikia\Search\Config must be set in the dependency container at a minimum in order to instantiate a result set.' );
+			throw new \Exception(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS112)